testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -587,77 +619,44 @@ impl EventObserver {
             }
             (true, Value::Response(_)) => STATUS_RESP_POST_CONDITION,
             _ => {
-                if let TransactionOrigin::Stacks(inner_tx) = &tx {
-                    if let TransactionPayload::PoisonMicroblock(..) = &inner_tx.payload {
-                        STATUS_RESP_TRUE
-                    } else {
-                        unreachable!() // Transaction results should otherwise always be a Value::Response type
-                    }
-                } else {
-                    unreachable!() // Transaction results should always be a Value::Response type
+                let TransactionOrigin::Stacks(inner_tx) = tx else {
+                    unreachable!("Transaction results should always be a Value::Response type");
+                };
+                if !matches!(inner_tx.payload, TransactionPayload::PoisonMicroblock(..)) {
+                    unreachable!("Transaction results should always be a Value::Response type");
                 }
+                STATUS_RESP_TRUE
```

Comment:
I think something like this would make this section even more clear:

```rust
    (_, _, TransactionOrigin::Stacks(Transaction { payload: TransactionPayload::PoisonMicroblock(..), .. })) => {
        STATUS_RESP_TRUE
    }
    _ => {
        unreachable!("Unexpected transaction result type")
    }
```